### PR TITLE
core: utility method for visualizing worlds as PlantUML

### PR DESCRIPTION
**Pull request by el-u**
## What is this fixing or adding?

This adds a utility method that allows world developers to debug their world visually by generating a PlantUML diagram representing the Regions and their connections via Entrances. This can be used, e.g., to quickly identify unconncted Entrances or orphaned Regions.

Locations and the filled Items can be show in the diagram as well, so one can also use this to illustrate the fill state.

## How was this tested?

See screenshots.

## If this makes graphical changes, please attach screenshots.

- ALttP, with locations hidden:
alttp_locations_hidden

- ~~I don't know what world this is~~ Adventure, after `create_regions`:
idk_after_create_regions

- ALttP again, but after `pre_fill`:
alttp_after_pre_fill

- KH2, fill completed (right before `generate_output`):
kh2_after_fill

- No comment:
no_comment

- No comment either:
no_comment_2

### CODE REVIEW

**Comment by Silvris** [comment]
2nd image is of Adventure.

File: Utils.py [hunk 1]
```diff
@@ -843,3 +844,115 @@ def freeze_support() -> None:
...
     multiprocessing.freeze_support()
+
+
+def visualize_regions(root_region: Region, file_name: str, *,
+                      show_entrance_names: bool = False, show_locations: bool = True, show_other_regions: bool = True,
+                      linetype_ortho: bool = True) -> None:
+    """Visualize the layout of a world as a PlantUML diagram.
+
+    Positional arguments:
+    root_region -- The region from which to start the diagram from. (Usually the "Menu" region of your world)
```
Comment (alwaysintreble) [inline Utils.py:855]: can we use the :param: keyword so IDEs pick this up?

**Comment by kindasneaki** [comment]
It might be a good idea to have something in the docs to explain what each thing does

**Review by alwaysintreble — APPROVED** [review]

**Comment by BootsinSoots** [comment]
I would like this very much. I am actually stuck at a point when generating where something like this would be very useful - I'm having problems locating exactly where my generation is getting extra locations from.

**Review by kindasneaki — APPROVED** [review]
I've ran this a couple times to test my world. It took me a bit to figure out how to open the .puml file but after doing so it was very helpful to see how all my regions connect together.

**Comment by Zunawe** [comment]
An additional thing that I would find helpful is being able to pass in a subset of regions or region names to highlight in a certain color or put in a bounding box or something. That way you could pass in `reachable_regions` from a `CollectionState` to visually see which regions are in sphere 1, or double check which regions open up when certain items are collected, or show the current state of your world when stepping through `fill_restrictive`, etc...

```py
state = CollectionState(self.multiworld)
state.collect(important_items[0])
state.collect(important_items[1])
...
visualize_regions(self.multiworld.get_region("Menu", self.player), "reachable_regions.puml",
                  highlight_regions=maximum_exploration_state.reachable_regions[self.player])
```

**Comment by Zunawe** [comment]
Fair enough. It's still very useful as-is.

**Review by black-sliver — APPROVED** [review]

**Pull request merged**